File: smb:/192.168.99.130/staging/Release/System/Data/Models/Ticket.js
var store = require("../Storage");
var user = require("./User");
var device = require("./Device");
exports.Table = undefined;
var Ticket = (function () {
    function Ticket() {
        this.uid = "";
        this.expire = 0;
        this.device_uid = "";
        this.owner_uid = "";
        this.attributes = {};
        this.accessTime = new Date();
    }
    Ticket.table = function () {
        if (!exports.Table) {
            exports.Table = store.DefineTable("Ticket", Ticket, { id: ["uid"] });
            Ticket.table()["hasOne"]('owner', user.User.table(), {
                required: true,
                autoFetch: true
            });
            Ticket.table()["hasOne"]('device', device.Device.table(), {
                required: true,
                autoFetch: true
            });
        }
        return exports.Table;
    };
    return Ticket;
})();
exports.Ticket = Ticket;

File: System/DB/Models/Ticket.lua
require 'Coat.Persistent'
require 'Coat.Persistent'.trace = print
local uuid = require 'uuid'

persistent "Ticket"
has_p.uid = { is = 'rw', isa = 'string' }
has_p.expire = { is = 'rw', isa = 'number' }
has_p.device_uid = { is = 'rw', isa = 'string' }
has_p.owner_uid = { is = 'rw', isa = 'string' }
has_p.attributes = { is = 'rw', isa = 'string' }
has_p.accessTime = { is = 'rw', isa = 'string' }

sql_create = [[
    CREATE TABLE Ticket (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        uid CHAR(64),
        expire INTEGER,
        device_uid CHAR(64),
        owner_uid CHAR(64),
        attributes TEXT,
        accessTime CHAR(24)
    )
]]

local conn = Ticket.establish_connection('sqlite3', "orm.db")
conn:execute(Ticket.sql_create)

local exports = {}
exports.Table = Ticket
return exports
